src/engraving/dom/lyrics.cpp

```diff
@@ -171,6 +171,15 @@ void Lyrics::layout2(int nAbove)
     LayoutData* ldata = mutLayoutData();
     double lh = lineSpacing() * style().styleD(Sid::lyricsLineHeight);
 
+    if ((staffType() && staffType()->xmlName() == "stdJianpu")
+        && (staffType() && staffType()->lines() == 0)) {
```

Comment:
I'm not sure if it's good to check `staffType()->lines() == 0`. If lines are never relevant in Jianpu notation, we should just ignore `lines()` for Jianpu and draw no staff lines regardless of the `lines()` property. 

(Also note that you're checking _twice_ if `staffType()` is not null; once is enough)